### DISCUSSION TITLE
how to get data from query params

### DISCUSSION THREAD

**Discussion by totaland** [vercel/next.js#25021] [Ideas]
I have a page that is built and exported to HTML and upload to the S3 bucket.

A) The name of the page is somelink.tsx

When I try to go to the link with this URL https://www.somelink?user=abc, the link is changed by NextJS to https://www.somelink/ 

So I can't query for the user.

B) The name of the page is [...user].tsx

When I go to the page, the URL was changed to https://somelink/?user=abc then it got redirect to the error page as the page does not exist....

How do I fix this issue?

**Comment by bobarros** [comment]
With no information about your code, how can the community help you? There are infinite possibilities.

**Reply by totaland** [reply]
Thanks mate

**Reply by totaland** [reply]
Build a NEXTJS project, upload it to s3 bucket try to directly access a page with query params with trailing slash turn on in next.config.js, then you will understand what my question is.

**Comment by totaland** [comment]
I found the answer, it is the NextJS export to HTML issues, the link to the page was  https://www.somelink?user=abc then when it is build and export to HTML the folder structure turn into somelink/index.html 
That is why when you directly go to the  https://www.somelink?user=abc it won't understand that the page exists as the somelink.tsx has been turn into index.html. So by changing the  https://www.somelink?user=abc to  https://www.somelink/index.html?user=abc then it works. That is very bad and it is not documented in any where in the docs.
